database/config.go

```diff
@@ -298,10 +298,7 @@ func (i *IndexConfig) ScanDocument(d document.Document) error {
 type Index struct {
 	index.Index
 
-	IndexName string
-	TableName string
-	Path      document.ValuePath
-	Unique    bool
+	Opts IndexConfig
```

Comment:
First of all thanks for doing this! 

However, I suggest to embed directly `IndexConfig`:
```go
type Index struct {
    index.Index
    IndexConfig
}
```
like that instead of accessing the config fields through `Opts`
```go
idx.Opts.Path.GetValue(d)
```
we can directly do
```go
idx.Path.GetValue(d)
````

What do you think?

Reply:
Actually, I am sure its a good idea because it will expose all method of the `IndexConfig`. What do you think @asdine ?

Reply:
Very good thing this avoids the refactoring of the majority of files.

Reply:
Yeah I would keep it like this, embedding would make the `Index` type API weird